globals.go

```diff
@@ -415,8 +415,9 @@ var (
 
 	// CanRunTaskStatus is a list of all host statuses in which a host could be running a task.
 	CanRunTaskStatus = []string{
```

Comment:
I would also filter the query that relies on this to check that the host is provisioned with user data, because the other queries that allow user data hosts that are not provisioned to run tasks always verify if it's a user data host. All non-user data hosts can only run tasks if they're running.

Reply:
I'm not sure that that's necessary, and makes the query significantly more complex. I'd like to avoid putting a lot of the user-data provisioning logic outside of actual user-data provisioning. CanRunTaskStatus keeps it fairly narrow.

Reply:
I meant that all the queries within the next task all check for user data hosts and their statuses and that's definitely outside the scope of host provisioning.

Somebody in the future going to read this and form the wrong conclusion about host statuses and running tasks, but I don't care very much.

Reply:
I want to think more about fixing this entirely, but I'm not sure what that looks like.